suitcase/mongo_embedded/__init__.py

```diff
@@ -0,0 +1,495 @@
+    from the RunEngine the volatile data is "frozen" by moving it to the
+    permanent database.
+
+    This Serializer ensures that when the stop document or close request is
+    received all documents we be written to the volatile database. After
+    everything has been successfully writen to volatile database, it will copy
+    all of the runs documents to the permanent database, check that it has been
+    correctly copied, and then delete the volatile data.
+
+    This Serializer assumes that documents have been previously validated
+    according to the bluesky event-model.
+
+    Note that this Seralizer does not share the standard Serializer
+    name or signature common to suitcase packages because it can only write
+    via pymongo, not to an arbitrary user-provided buffer.
+
+    Examples
+    --------
+    >>> from bluesky import RunEngine
+    >>> from bluesky.plans import scan
+    >>> from mongobox import MongoBox
+    >>> from ophyd.sim import det, motor
+    >>> from suitcase.mongo_embedded import Serializer
+
+    >>> # Create two sandboxed mongo instances
+    >>> volatile_box = MongoBox()
+    >>> permanent_box = MongoBox()
+    >>> volatile_box.start()
+    >>> permanent_box.start()
+
+    >>> # Get references to the mongo databases
+    >>> volatile_db = volatile_box.client().db
+    >>> permanent_db = permanent_box.client().db
+
+    >>> # Create the Serializer
+    >>> serializer = Serializer(volatile_db, permanent_db)
+
+    >>> RE = RunEngine({})
+    >>> RE.subscribe(serializer)
+
+    >>> # Generate example data.
+    >>> RE(scan([det], motor, 1, 10, 10))
+    """
+
+    def __init__(self, volatile_db, permanent_db, num_threads=1,
+                 buffer_size=1000000, page_size=5000000, **kwargs):
+
+        """
+        Insert documents into MongoDB using an embedded data model.
+
+        Parameters
+        ----------
+        volatile_db: pymongo database
+            database for temporary storage
+        permanent_db: pymongo database
+            database for permanent storage
+        num_theads: int, optional
+            number of workers that read from the buffer and write to the
+            database. Must be 5 or less. Default is 1.
+        buffer_size: int, optional
+            maximum size of the buffers in bytes. buffer_size + page_size must
+            be less than 15000000. Default is 5000000
+        page_size: int, optional
+            The document size for event_page and datum_page documents. The
+            maximum event/datum_page size is buffer_size + page_size.
+            buffer_size + page_size must be less than 15000000.The
+            default is 5000000.
+        """
+
+        if num_threads > 10 or num_threads < 1:
+            raise AttributeError(f"num_threads must be between 1 and 5 "
+                                 "inclusive.")
+
+        if page_size < 10000:
+            raise AttributeError(f"page_size must be >= 10000")
```

Comment:
Why? It might be useful to allow small pages sizes for testing.

Reply:
good point, I updated it